src/timer.c

```diff
@@ -408,14 +433,16 @@ void restart_timer(timer_link_t *timer)
 {
 	if (timer)
 	{
+		disable_interrupts();
+
+		uint32_t now = TICKS_READ();
 		timer->left = TICKS_READ() + (int32_t)timer->set;
```

Comment:
```suggestion
		timer->left = now + (int32_t)timer->set;
```
not that it matters I think but while you're at it